crates/valence/examples/chat.rs

```diff
@@ -0,0 +1,115 @@
+        .add_plugin(ServerPlugin::new(()))
+        .add_startup_system(setup)
+        .add_system_to_stage(EventLoop, default_event_handler)
+        .add_system_to_stage(EventLoop, handle_message_events)
+        .add_system_to_stage(EventLoop, handle_command_events)
+        .add_system(init_clients)
+        .add_system(despawn_disconnected_clients)
+        .add_system_set(PlayerList::default_system_set())
+        .add_system(teleport_oob_clients)
+        .run();
+}
+
+fn setup(world: &mut World) {
+    let mut instance = world
+        .resource::<Server>()
+        .new_instance(DimensionId::default());
+
+    for z in -5..5 {
+        for x in -5..5 {
+            instance.insert_chunk([x, z], Chunk::default());
+        }
+    }
+
+    // Create circular arena.
+    for z in -ARENA_RADIUS..ARENA_RADIUS {
+        for x in -ARENA_RADIUS..ARENA_RADIUS {
+            let dist = f64::hypot(x as _, z as _) / ARENA_RADIUS as f64;
+
+            if dist > 1.0 {
+                continue;
+            }
+
+            let block = if rand::random::<f64>() < dist {
+                BlockState::STONE
+            } else {
+                BlockState::DEEPSLATE
+            };
+
+            for y in 0..SPAWN_Y {
+                instance.set_block_state([x, y, z], block);
+            }
+        }
+    }
+
+    world.spawn(instance);
+}
+
+fn init_clients(
+    mut clients: Query<&mut Client, Added<Client>>,
+    instances: Query<Entity, With<Instance>>,
+) {
+    for mut client in &mut clients {
+        client.set_position([0.0, SPAWN_Y as f64, 0.0]);
+        client.set_instance(instances.single());
+        client.set_game_mode(GameMode::Adventure);
+        client.send_message("Welcome to Valence! Talk about something.".italic());
+    }
+}
+
+fn handle_message_events(mut clients: Query<&mut Client>, mut messages: EventReader<ChatMessage>) {
+    for message in messages.iter() {
+        let Ok(client) = clients.get_component::<Client>(message.client) else {
+            warn!("Unable to find client for message: {:?}", message);
+            continue;
+        };
+
+        let message = message.message.to_string();
+
+        let formatted = format!("<{}>: ", client.username())
+            .bold()
+            .color(Color::YELLOW)
+            + message.into_text().not_bold().color(Color::WHITE);
+
+        clients.par_for_each_mut(16, |mut client| {
+            client.send_message(formatted.clone());
+        })
+    }
+}
+
+fn handle_command_events(mut clients: Query<&mut Client>, mut commands: EventReader<ChatCommand>) {
+    for command in commands.iter() {
+        let Ok(mut client) = clients.get_component_mut::<Client>(command.client) else {
+            warn!("Unable to find client for message: {:?}", command);
+            continue;
+        };
+
+        let message = command.command.to_string();
+
+        let formatted =
+            "You sent the command ".into_text() + ("/".into_text() + (message).into_text()).bold();
```

Comment:
I think would be good to have an example of command like `/echo`.

Reply:
That is literally what this command is, in my opinion though, Chat Example != Command Example, and I think the command code doesn't (shouldn't) be in this example. Especially since a proper command system is yet to be built. and when it does, the more examples that uses/demonstrates commands will have to be changed, so to reduce the overhead on that, I would strongly suggest removing it from this example.

Reply:
No, this is perfectly fine. We don't have command parsing implemented yet. An example implementing commands would go in a separate example.

Reply:
I agree with @dyc3 for this - even if I added a (complex) command system, it would be incomplete until brigadier nodes are sent to the client, and I hardly even know where to start with that other than #35.

Reply:
For reference: we have a thread on discord discussing the design for a `valence_command` crate: https://discord.com/channels/998132822239870997/1072277129183625256